src/searchReplace.tsx

```diff
@@ -306,6 +356,32 @@ const SearchReplaceElement = (props: IProps) => {
         />
         {props.children}
       </div>
+      <div>
+        <TextField
+          appearance="outline"
+          placeholder="files to"
+          onInput={(event: any) => {
+            const excludeToggle = document
+              .getElementById('exclude-toggle')
+              ?.getAttribute('current-checked');
+            props.onFileFilter(event.target.value, excludeToggle === 'true');
+          }}
+        >
+          File filters
+        </TextField>
+        <Switch
+          id="exclude-toggle"
+          onClick={() => {
+            const excludeToggle = document
+              .getElementById('exclude-toggle')
+              ?.getAttribute('current-checked');
+            props.onExcludeToggle(excludeToggle === 'true');
+          }}
```

Comment:
This is not the React way.

You need to pass the model `excludeToggle` variable in the Search component that will set the attribute `checked` on the switch.

And you should use `onChange` not `onClick`.